lab_2_retrieval_w_bm25/main.py

```diff
@@ -18,6 +18,25 @@ def tokenize(text: str) -> list[str] | None:
 
     In case of corrupt input arguments, None is returned.
     """
+    if not isinstance(text, str):
+        return None
+    word_list = text.split(' ')
+    for word_num in range(len(word_list)):
```

Comment:
why so many for loops?? its just a tokenize function. I will accept this code only if this function will have 1 for loop.

Reply:
I have rewritten this function